platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/snapshot/StateSnapshotManagerMetrics.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.snapshot;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.metrics.RunningAverageMetric;
+import com.swirlds.metrics.api.Counter;
+import com.swirlds.metrics.api.Metrics;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Encapsulates metrics for the state snapshot manager.
+ */
+public class StateSnapshotManagerMetrics {
+
+    private static final RunningAverageMetric.Config WRITE_STATE_TO_DISK_TIME_CONFIG = new RunningAverageMetric.Config(
+                    "platform", "writeStateToDisk")
+            .withDescription("average time it takes to write a SignedState to disk (in milliseconds)")
+            .withUnit("ms");
+
+    private final RunningAverageMetric writeStateToDiskTime;
+
+    private static final RunningAverageMetric.Config STATE_TO_DISK_TIME_CONFIG = new RunningAverageMetric.Config(
+                    "platform", "stateToDisk")
+            .withDescription("average time it takes to do perform all actions when writing a SignedState to disk "
+                    + "(in milliseconds)")
+            .withUnit("ms");
+    private final RunningAverageMetric stateToDiskTime;
+
+    private static final Counter.Config TOTAL_NEVER_SIGNED_STATES_CONFIG = new Counter.Config(
+                    "platform", "totalNeverSignedStates")
+            .withDescription("total number of states that did not receive enough signatures in the allowed time")
+            .withUnit("count");
+    private final Counter totalNeverSignedStates;
```

Comment:
You put this metric in the new class, but it's unused here, and still exists in the original `SignedStateMetrics` class

Reply:
Good catch. Technically harmless (second registration gets linked to the same metric), but not good to have. Removed.